File: java/WriteToNamedPipe.java
import java.nio.charset.StandardCharsets;
import java.nio.file.*;

public class WriteToNamedPipe {
    public static void main(String[] args) throws Exception {
        var outFile = Paths.get("output.pipe");
        var text = "Hello World!";
        Files.write(outFile, text.getBytes(StandardCharsets.UTF_8));
    }
}

File: cplusplus/write_to_named_pipe.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::string outFile = "output.pipe";
    std::string text = "Hello World!";

    std::ofstream file(outFile, std::ios::out | std::ios::binary);

    file.write(text.c_str(), text.size());
    file.close();

    return 0;
}
